### DISCUSSION TITLE
Latex `aligned` in typst

### DISCUSSION THREAD

**Discussion by qres** [typst/typst#3879] [Q&A] [answered]
Is there a good way to get the effect of the `aligned` environment in latex?

For example in the following Latex equation, the plus signs in the second right hand side are vertically aligned.
```
\begin{align*}
    A &= b \\
    &= c \begin{aligned}[t]
        & + d \\
        & + e + f \\
        & + g + h
    \end{aligned}
\end{align*}
```

I tried the following in typst, but getting the baseline right in the following code is rather difficult:
```
$
  A &= b \
    &= c #box(baseline: 100% - 0.75em, $
      &#math.class("binary", [+]) d \
      &#math.class("binary", [+]) e + f \
      &#math.class("binary", [+]) g + h
    $) 
$
```
![Screenshot from 2024-04-05 12-41-36](https://github.com/typst/typst/assets/7228334/5611d31a-7e88-48d4-9c38-824c324036f9)

**Comment by EpicEricEE** [comment]
You don't really need the box, you can have multiple alignment points in the equation. If that works for you, you can do something like
```typ
#show math.plus: math.class.with("binary") // So you don't need to specifiy it every time

$
  A &= b \
    &= c && + d \
    &    && + e + f \
    &    && + g + h
$
```
The `&&` is necessary to make it left aligned, as the alignment switches between right and left after every `&` symbol.

**Reply by qres** [reply]
Thank you for the suggestion! Unfortunately, it seems like I reduced my problem a bit too much. When the first right hand side is longer than $c$, this leads to additional space between $c$ and $+$:

```
#show math.plus: math.class.with("binary")

$
  A &= sin(b) \
    &= c && + d \
    &    && + e + f \
    &    && + g + h
$
```

**Comment by EpicEricEE** [comment, marked as answer]
In case you're still interested, maybe this is a better solution:
```typ
#show math.plus: math.class.with("binary")

$
  A & = sin(b) \
    & #block[$
      = c &+ d \
          &+ e + f \
          &+ g + h
    $]
$
```

If you have numbered equations, you need to add this show rule to prevent the inner equation from being numbered:
```typ
#show math.equation.where(block: true): it => {
  set math.equation(numbering: none)
  it
}
```

![image](https://github.com/typst/typst/assets/7191192/dd5c9401-ea51-463a-ac26-06a282fc54ef)

**Reply by qres** [reply]
Thank you, this helps a lot! Also the hint about the numbering.
